src/ActivityRepository.js

```diff
@@ -6,7 +6,7 @@ class ActivityRepository {
   calculateAllUserDailyAverage(day, type) {
     let users = [];
     this.allUsersActivity.forEach(entry => {
-      if (!users.includes(entry.userID)) {
+      if (!users.includes(entry.userID) && entry.date === day) {
```

Comment:
Ah, I see you resolved the issue expressed in my previous comment by making membership in user array conditional by date...Still, I think we can shorten this method by using the same iterator that sums data to tally relevant entries